src/main/java/org/fogbowcloud/saps/engine/core/scheduler/Scheduler.java

```diff
@@ -373,7 +404,7 @@ else if (state == ImageTaskState.PREPROCESSING)
      * submitted jobs to Arrebol.
      */
     private void checker() {
-        List<JobSubmitted> submittedJobs = arrebol.returnAllJobsSubmitted();
+        List<JobSubmitted> submittedJobs = this.submittedJobsID;
```

Comment:
Do not create a new collection, use the `submittedJobs` attribute of this class

Reply:
Done.